Extensions/xkit_preferences.js

```diff
@@ -394,7 +412,7 @@ XKit.extensions.xkit_preferences = new Object({
 
 		},
 
-		check: function(id) {
+		check_for_news_existing: function(id, version) {
```

Comment:
I dislike returning "magic numbers" (see L429, L431, L437), but if there's no other way to represent multiple states, please at least document the function and its return values.